src/os/bluestore/bluestore_tool.cc

```diff
@@ -219,6 +227,54 @@ void inferring_bluefs_devices(vector<string>& devs, std::string& path)
   }
 }
 
+void bluefs_import(
+  string input_file,
+  CephContext *cct,
+  const string& path,
+  const vector<string>& devs)
+{
+  int r;
+  std::ifstream f(input_file.c_str(), std::ifstream::binary);
+  if (!f) {
+    r = -errno;
+    cerr << "open " << input_file.c_str() << " failed: " << cpp_strerror(r) << std::endl;
+    exit(EXIT_FAILURE);
+  }
+
+  std::unique_ptr<BlueFS> bs = std::unique_ptr<BlueFS>(open_bluefs(cct, path, devs));
+
+  const string dir = "db";
+  BlueFS::FileWriter *h;
+  fs::path file_path(input_file);
+  const string file_name = file_path.filename();
+  r = bs->open_for_write(dir, file_name, &h, true);
+  if (r < 0) {
+    cerr << "Open " << dir << "/" << file_name << " failed: " << cpp_strerror(r) << std::endl;
+    exit(EXIT_FAILURE);
+  }
+  uint64_t dsize = 0;
+  r = bs->stat(dir, file_name, &dsize, NULL);
+  if (r != 0){
+    cerr << "Get " << dir << "/" << file_name << " stat failed: " << cpp_strerror(r) << std::endl;
+    exit(EXIT_FAILURE);
+  }
+  uint64_t fsize = fs::file_size(input_file.c_str());
+  if (fsize < dsize){
+      bs->truncate(h, uint64_t(fsize));
+  }
+  string buffer;
+  while (!f.eof()) { 
+    getline(f, buffer);
+    h->append(buffer.c_str(), strlen(buffer.c_str()));
```

Comment:
suggest to treat files as 100% binary - please avoid strlen() and getline() calls